File: 01_Language/01_Functions/go/src/functions/is_link.go
package functions

import (
	"os"
)

func IsLink(filename string) bool {
	lstat, err := os.Lstat(filename)
	if err != nil {
		return false
	}
	if lstat.Mode() & os.ModeSymlink != 0 {
		return true
	}
	return false
}

File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/IsLink.java
package com.onevgo.functions;

import org.apache.commons.io.FileUtils;

import java.io.File;
import java.io.IOException;

public class IsLink {
    public static boolean isLink(String filename) {
        try {
            return FileUtils.isSymlink(new File(filename));
        } catch (IOException e) {
            e.printStackTrace();
        }
        return false;
    }

    public static void main(String[] args) {
        System.out.println(isLink("test.txt"));
    }
}
